zebra-rpc/src/methods.rs

```diff
@@ -601,6 +608,28 @@ where
                     _ => unreachable!("unmatched response to a block request"),
                 }
             } else if verbosity == 1 {
+                // # Performance
+                //
+                // This RPC is used in `lightwalletd`'s initial sync of 2 million blocks,
+                // so it needs to load all its fields very efficiently.
+                //
+                // Currently, we get the transaction IDs from an index, which is much more
+                // efficient than loading all the block data and hashing all the transactions.
+
+                // TODO: look up the hash if we only have a height,
+                //       and look up the height if we only have a hash
+                let hash = hash_or_height.hash().map(GetBlockHash);
+                let height = hash_or_height.height();
+
+                // TODO: do these state  queries in parallel?
```

Comment:
```suggestion
                // TODO: do these state queries in parallel?
```

just an extra space here, not important at all.

Reply:
I'll fix this in another PR so we can unblock mining testing.

Reply:
Related: do we want to open an issue for running the fetch_mempool_transactions and fetch_state_tip_and_local_time queries in the get_block_template method or any other requests in parallel too?

Reply:
We usually leave optional cleanups and minor performance fixes as TODO comments, to keep our issue tracker focused.

If there is a specific user need, we can go ahead and open an issue. But for now I think we're so much faster than `zcashd` here that no-one will ever notice.